### PR TITLE
refactor(http): Import proper OutgoingHttpHeaders on all http clients

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Fix imports, remove `unknown` type, remove TODO item.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/http/index.ts [hunk 1]
```diff
@@ -9,17 +9,13 @@ import { resolveBaseUrl } from '../url';
 import { applyAuthorization, removeAuthorization } from './auth';
 import { applyHostRules } from './host-rules';
 import { getQueue } from './queue';
-import { GotOptions, RequestStats } from './types';
+import type { GotOptions, OutgoingHttpHeaders, RequestStats } from './types';
 
 // TODO: refactor code to remove this
 import './legacy';
 
 export * from './types';
```
Comment (viceice) [inline lib/util/http/index.ts:17]: This should be removed too. Can be separate pr.

File: lib/util/http/types.ts [hunk 2]
```diff
@@ -23,3 +23,7 @@ export interface RequestStats {
   duration: number;
   queueDuration: number;
 }
+
+export interface OutgoingHttpHeaders {
```
Comment (viceice) [inline lib/util/http/types.ts:27]: We should extend from `Record` here

**Comment by zharinov** [comment]
Deep merge of `got` options didn't affect any tests and somehow related to headers. So, pushing one more commit here.

File: lib/util/http/types.ts [hunk 3]
```diff
@@ -23,3 +23,7 @@ export interface RequestStats {
   duration: number;
   queueDuration: number;
 }
+
+export interface OutgoingHttpHeaders extends Record<string, unknown> {
+  [header: string]: string | string[] | undefined;
```
Comment (viceice) [inline lib/util/http/types.ts:28]: Remove and add this types to second record type

Comment (zharinov) [reply lib/util/http/types.ts]: Replaced by `Record<string, string | string[] | undefined>`, because empty interfaces don't lint well

Comment (viceice) [reply lib/util/http/types.ts]: Better use a type def then?

File: lib/util/http/types.ts [hunk 4]
```diff
@@ -23,3 +23,5 @@ export interface RequestStats {
   duration: number;
   queueDuration: number;
 }
+
+export type OutgoingHttpHeaders = Record<string, string | string[] | undefined>;
```
Comment (zharinov) [inline lib/util/http/types.ts:27]: Okay, just commit your suggestion 🙃

**Review by viceice — APPROVED** [review]
